### PR TITLE
[DSLX:LSP] Add basic support for inlay hints on let bindings.

### PR DESCRIPTION
**Pull request by cdleary**
Was dependent on https://github.com/chipsalliance/verible/pull/2241, which is now submitted.

### CODE REVIEW

File: xls/dslx/lsp/lsp_type_utils.cc [hunk 1]
```diff
@@ -22,13 +22,15 @@
 
 namespace xls::dslx {
 
+verible::lsp::Position ConvertPosToLspPosition(const Pos& pos) {
```
Comment (hzeller) [inline xls/dslx/lsp/lsp_type_utils.cc:25]: +1 for the readability improvement this one brings below.

**Comment by cdleary** [comment]
Pointed at Verible's latest commit and git-clang-format'd, should be all set now.

**Review by hzeller — COMMENTED** [review]
LGTM. Just change the reference in the load_external.bzl once the merged change over in verible dropped the release.

File: dependency_support/load_external.bzl [hunk 2]
```diff
@@ -295,12 +295,12 @@ def load_external_repositories():
...
     )
 
-    # Tagged 2024-06-19, current as of 2024-06-26
+    # Current as of 2024-08-23
     http_archive(
         name = "verible",
-        sha256 = "8c5d13bdc4f4bc756b0f6d31b61cc8aba75dfcfcf22822244ac9c0ccb43db814",
-        strip_prefix = "verible-0.0-3704-g1d393d43",
-        urls = ["https://github.com/chipsalliance/verible/archive/refs/tags/v0.0-3704-g1d393d43.tar.gz"],
+        sha256 = "0f6e2f0cad335c9aca903aaa014f2a55d40c5c4dbfe4666474771f5c08e4a42c",
```
Comment (hzeller) [inline dependency_support/load_external.bzl:301]: Verible upstream with the needed change is 
https://github.com/chipsalliance/verible/releases/tag/v0.0-3754-g17e909b0

Comment (cdleary) [reply dependency_support/load_external.bzl]: Switched to use the tag instead of the commit.

**Review by hzeller — APPROVED** [review]
LGTM

File: dependency_support/load_external.bzl [hunk 3]
```diff
@@ -295,12 +295,13 @@ def load_external_repositories():
         urls = ["https://github.com/grailbio/bazel-compilation-database/archive/940cedacdb8a1acbce42093bf67f3a5ca8b265f7.tar.gz"],
     )
 
-    # Tagged 2024-06-19, current as of 2024-06-26
+    # Tagged 2024-08-23, current as of 2024-08-24
+    VERIBLE_TAG = 'v0.0-3756-gda9a0f8c'
```
Comment (hzeller) [inline dependency_support/load_external.bzl:299]: Looks like bazel [`buildifier`](https://github.com/bazelbuild/buildtools/blob/main/buildifier/README.md) prefers double quotes here (and below the 'v').

We should add running `buildifier` in the CI on BUILD and *.bzl files...

Comment (cdleary) [reply dependency_support/load_external.bzl]: Done on the double quotes.

**Pull request merged**